BlogTemplate/Pages/Edit.cshtml.cs

```diff
@@ -58,13 +58,31 @@ public IActionResult OnPostSaveDraft()
         public void UpdatePost(Post newPost, string slug)
         {
             oldPost = _dataStore.GetPost(slug);
+            DateTime OrigPubDate = oldPost.PubDate;
+            // Body and excerpt are not updated either. They are whatever they were inputted into the form as
+            newPost.PubDate = OrigPubDate;
             newPost.Tags = Request.Form["Tags"][0].Replace(" ", "").Split(",").ToList();
-            
-            SlugGenerator slugGenerator = new SlugGenerator(_dataStore);
-            newPost.Slug = slugGenerator.CreateSlug(newPost.Title);
+
+            if (Request.Form["updateslug"] == "true")
+            {
+                SlugGenerator slugGenerator = new SlugGenerator(_dataStore);
+                newPost.Slug = slugGenerator.CreateSlug(newPost.Title);
+            }
+
+            else
+            {
+                newPost.Slug = oldPost.Slug;
+            }
+
             newPost.Comments = oldPost.Comments;
 
             _dataStore.UpdatePost(newPost, oldPost);
         }
+
+        public IActionResult SlugUpdateWarning()
+        {
+            TempData["notice"] = "Updated slug";
+            return Redirect("/Edit");
```

Comment:
Not sure if this is happening here/not sure if this is what we want to happen, but when the alert pops up asking the user if they want to continue with changing the slug, if you click cancel it redirects to the edit page with the title changed but the slug unchanged. This is a slightly odd experience because usually when you click cancel on something you expect that your changes will not go through.